File: tests/interfile/cpp/test_hof_comprehensive_cpp_split/customMap.cpp
template<typename T>
std::vector<T> customMap(const std::vector<T>& arr, std::function<T(T)> callback) {
    std::vector<T> result;
    for (const auto& item : arr) {
        result.push_back(callback(item));
    }
    return result;
}

File: tests/interfile/ruby/test_hof_comprehensive_ruby_split/custom_map.rb
require_relative 'custom_for_each'
require_relative 'test_custom_map'
require_relative 'test_custom_foreach'
require_relative 'test_builtin_map'
require_relative 'test_builtin_each'
require_relative 'test_builtin_select'
require_relative 'source'
require_relative 'sink'
require_relative 'toplevel_handler'
require_relative 'zz_main'
def custom_map(arr, &callback)
  result = []
  arr.each do |item|
    result << callback.call(item)
  end
  result
end
